src/wasm/wasm-binary.cpp

```diff
@@ -2192,23 +2192,24 @@ void WasmBinaryBuilder::readNames(size_t payloadLen) {
       continue;
     }
     auto num = getU32LEB();
+    std::map<Name, int> nameCounts;
     for (size_t i = 0; i < num; i++) {
       auto index = getU32LEB();
       auto name = getInlineString();
+      // De-duplicate names by appending _1, _2, etc.
+      if (nameCounts[name] != 0) {
```

Comment:
I think you need a loop here, as when appending it might still be something already seen.

Reply:
At first I thought if you had say 3 `foo`s, you'd get `foo` `foo_1` and `foo_2` - which you would.
But the problem you're pointing out is if you had 2 `foo`s and a `foo_1`, which would break it.
This sounds like the perfect thing to add a test for...

Reply:
Isn't the llvm convention for this to add  ".1" ?   (i.e. isn't this was llmv-link does?)

At least this is what lld does wen you ask it to export locals and you have multiple locals with the same name (from different compilation units).

Reply:
https://github.com/llvm-mirror/lld/blob/master/wasm/Writer.cpp#L622

Reply:
This is also what wabt does, and I'll change this to do that as well so that it's less likely to have this sort of transitive collision.